pkg/dataplane/types.go

```diff
@@ -115,13 +116,24 @@ func mode(v3ioFileMode FileMode) os.FileMode {
 	const S_IFMT = 0xf000
 	const IP_OFFMASK = 0x1fff
 
-	// Convert 16 bit octal representation of V3IO into decimal 32 bit representation of Go
-	mode, err := strconv.ParseUint(string(v3ioFileMode), 8, 32)
-	if err != nil {
-		panic(err)
+	// Note, File mode from different API's has different base.
+	// For example Scan API returns file mode as decimal number (base 10) while ListDir as Octal (base 8)
+	var sFileMode = string(v3ioFileMode)
+	if strings.HasPrefix(sFileMode, "0") {
+		// Convert 16 bit octal representation of V3IO into decimal 64 bit representation of Go
+		if mode, err := strconv.ParseUint(sFileMode, 8, 64); err != nil {
+			panic(err)
```

Comment:
Better return an error. It's kind of scary that `String()` can cause a panic... It probably shouldn't.

Reply:
Why panic here? And why add error to the return signature if we panic rather than returning it?